.github/workflows/build.yaml

```diff
@@ -36,6 +36,11 @@ jobs:
         uses: github/codeql-action/init@v3
         with:
           languages: javascript-typescript, c-cpp
+      - name: Install system libcurl
+        if: matrix.preset == 'linux-arm64-ci' || matrix.preset == 'linux-ci'
+        run: |
+          sudo apt update
+          sudo apt install -y libcurl4-openssl-dev
```

Comment:
If we explicitly have to pick the openssl flavor that suggests something is broken. Is there a `libcurl4-dev`?

In particular using the platform's native TLS implementation is important; recent debian for instance defaults to a different one for HTTP3 support

Reply:
I just picked one arbitrarily, we need to get the headers from a dev package (any would do). Testing locally I get:

```
Package libcurl4-dev is a virtual package provided by:
  libcurl4-openssl-dev 8.5.0-2ubuntu10.6
  libcurl4-gnutls-dev 8.5.0-2ubuntu10.6
You should explicitly select one to install.
```